In [1]:
library(repr)
library(tidyverse)
library(tidymodels)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
── Attaching packages ────────────────────────────────────── tidymodels 1.1.1 ──

✔ broom        1.0.6     ✔ rsample      1.2.1
✔ dials        1.3.0     ✔ tune         1.1.2
✔ infer        1.0.7     ✔ workflows    1.1.4
✔ modeldata    1.4.0     ✔ workflowsets 1.0.1
✔ parsnip      1.2.1     ✔ yardstick    1.3.1
✔ recipes      1.1.0     

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ scales::discard() masks purrr::discard()
✖ dplyr::filt

In [3]:
players<-read_csv("https://raw.githubusercontent.com/vjia02/individual_assignment_github/refs/heads/main/players.csv")
head(players)
nrow(players)
summary_stats <- players |>
    select(played_hours, Age) |>
    map_dbl(mean, na.rm=TRUE)
summary_stats

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

[1] 196

played_hours          Age 
    5.845918    21.139175

(1) Data Description:
Provide a full descriptive summary of the dataset, any other potential issues related to things you cannot directly see.

Note that the selected dataset(s) will probably contain more variables than you need. In fact, exploring how the different variables in the dataset affect your model may be a crucial part of the project. You need to summarize the full data regardless of which variables you may choose to use later on.

The project was conducted by the Dr. Frank Wood's computer science research team at UBC, where they created a Minecraft server for members of the public to play. The data was collected from players' accounts when registering and also each of their played sessions.

The players.csv dataset has 196 observations and 7 variables. See below for a table of the variables. An issue I noticed is that the Age variable is double type, while I believe it should be an integer type because age is counted in whole years.
potential issues related to things i can't directly see:

The sessions.csv dataset
I'm not using the sessions.csv dataset because the question I want to answer will only require the total amount of time they play, and not the duration of each session or the exact time of day in which they play.

|Variable| Variable type | Description|Mean Value|Dataset|
|:---------:|:-----------:|:----:|:-----:|:----:|
|experience| character |Whether the player is an Amateur, Regular, Veteran, or Pro|N/A|players|
|subscribe| logical| Whether or not the player is subscribed to the server|N/A|players|
|hashedEmail| character| Unique email of the player|N/A|players and sessions|
|played_hours| double| Number of hours played|5.85|players|
|name| character| Player's first name|N/A|players|
|gender| character| Player's gender|N/A|players|
|Age| double| Player's age in years|21.14|players|
|start_time|
|end_time|

(2) Questions:
Describe clearly how the data will help you address the question of interest. You may need to describe how you plan to wrangle your data to get it into a form where you can apply one of the predictive methods from this class.

(2) Questions:
The broad question that I will address is: "What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?" More specifically, can the number of played hours or the age of the player predict whether or not a player is subscribed to the newsletter in the players dataset? 
I will first look for a general correlation between the hours played and age variables and the subscription variable. Then, I plan to group the players by their skill level (player type) and look for individual trends there as well.

(3) Exploratory Data Analysis and Visualization
In this assignment, you will:

**Demonstrate that the dataset can be loaded into R.**
**Do the minimum necessary wrangling to turn your data into a tidy format. Do not do any additional wrangling here; that will happen later during the group project phase.
Compute the mean value for each quantitative variable in the players.csv data set. Report the mean values in a table format.**
Make a few exploratory visualizations of the data to help you understand it.
Use our visualization best practices to make high-quality plots (make sure to include labels, titles, units of measurement, etc)
Explain any insights you gain from these plots that are relevant to address your question
Note: do not perform any predictive analysis here. We are asking for an exploration of the relevant variables to demonstrate that you understand them well before performing any additional modelling, and to identify potential problems you anticipate encountering.






(3) The players.csv dataset is already in a tidy format, and we can tell because each row is a single observation (a single player's variables), each column is a single variable (one and only one variable per column), and each cell is a single value, not multiple.

(4) Methods and Plan

Propose one method to address your question of interest using the selected dataset and explain why it was chosen. Do not perform any modelling or present results at this stage. We are looking for high-level planning regarding model choice and justifying that choice.

In your explanation, respond to the following questions:

Why is this method appropriate?
Which assumptions are required, if any, to apply the method selected?
What are the potential limitations or weaknesses of the method selected?
How are you going to compare and select the model?
How are you going to process the data to apply the model? For example: Are you splitting the data? How? How many splits? What proportions will you use for the splits? At what stage will you split? Will there be a validation set? Will you use cross validation?

(4)
- plan to use knn classification for this
